compute_tools/src/bin/compute_ctl.rs

```diff
@@ -238,10 +238,25 @@ fn main() -> Result<()> {
         thread::sleep(Duration::from_secs(30));
     }
 
-    info!("shutting down tracing");
     // Shutdown trace pipeline gracefully, so that it has a chance to send any
-    // pending traces before we exit.
-    tracing_utils::shutdown_tracing();
+    // pending traces before we exit. Shutting down OTEL tracing provider may
+    // hang for quite some time, see, for example:
+    // - https://github.com/open-telemetry/opentelemetry-rust/issues/868
+    // - and our problems with staging https://github.com/neondatabase/cloud/issues/3707#issuecomment-1493983636
+    //
+    // Yet, we want computes to shut down fast enough, as we may need a new one
+    // for the same timeline ASAP. So wait no longer than 2s for the shutdown to
+    // complete, then just error out and exit the main thread.
+    info!("shutting down tracing");
+    let (sender, receiver) = mpsc::channel();
+    let _ = thread::spawn(move || {
+        tracing_utils::shutdown_tracing();
+        if let Ok(()) = sender.send(true) {}
```

Comment:
```suggestion
        sender.send(()).ok()
```

Some style nits:

We don't care about other half preliminary drop since it never happens and that's the only error that send can return, so `ok()` it.

Then, send `()` instead of `bool` since we don't care about the result actually, only the event arrival.

Reply:
Thanks! It felt odd to send data that nobody checks, but I didn't know that I can send 'nothing'. Something got broken in the remote e2e, so will wait merging it until e2e pass